backend/src/tasks/LookingGlass.ts

```diff
@@ -0,0 +1,286 @@
+    .leftJoinAndSelect('domain.organization', 'organization')
+    .andWhere('ip IS NOT NULL');
+
+  if (organizationId) {
+    pulledDomains = pulledDomains.andWhere('domain.organization=:org', {
+      org: organizationId
+    });
+  }
+  const D = await pulledDomains.getMany();
+  return D;
+}
+
+export const handler = async (commandOptions: CommandOptions) => {
+  const { organizationId, organizationName, scanId } = commandOptions;
+
+  const Org = await getOrg(organizationId);
+
+  console.log('Running lookingGlass on organization', organizationName);
+
+  const collections: LGCollectionsResponse[] = await collectionByWorkspace();
+  const ipsAndDomains: string[] = [];
+  const domains: Domain[] = [];
+  const Vulns_list: any = [];
+  const Vulnerabilities: Vulnerability[] = [];
+
+  for (const line of collections) {
+    //Match the organization to the LookingGlass Collection
+    if (organizationName == line['name']) {
+      const collectionID = line['id'];
+      console.log(line['name']);
+      //Query LookingGlass for the Threat info
+      const data: LGThreatResponse[] = await getThreatInfo(collectionID);
+      const responseDomains: Domain[] = await saveAndPullDomains(
+        data,
+        organizationId,
+        scanId,
+        Org
+      );
+      for (const l of data['results']) {
+        //Create a dictionary of relevant fields from the API request
+        if (typeof Org === 'object') {
+          const val = {
+            firstSeen: new Date(l['firstSeen']),
+            lastSeen: new Date(l['lastSeen']),
+            sources: l['sources'],
+            ref_type: l['ref']['type'],
+            ref_right_type: l['ref']['right']['type'],
+            ref_right_id: l['ref']['right']['id'],
+            ref_left_type: l['ref']['left']['type'],
+            ref_left_id: l['ref']['right']['id'],
+
+            right_ticScore: l['right']['ticScore'],
+            right_classifications: l['right']['classifications'],
+            right_name: l['right']['name'],
+
+            left_type: l['left']['type'],
+            left_ticScore: l['left']['ticScore'],
+            left_name: l['left']['name']
+          };
+          if (l['right']['classifications'][0] == 'Vulnerable Service') {
+            val['vulnOrMal'] = 'Vulnerability';
+          } else {
+            val['vulnOrMal'] = 'Malware';
+          }
+
+          //if the Domain already exists and add this vuln to the associated vulnerability
+          if (ipsAndDomains.includes(l['left']['name'])) {
+            for (const Vuln of Vulns_list) {
+              if (l['left']['name'] == Vuln['domain'].name) {
+                Vuln['structuredData']['lookingGlassData'].push(val);
+              }
+            }
+          }
+          //if the Domain hasn't been used create a new Domain and a new Vulnerability
+          else {
+            for (const x of responseDomains) {
+              if (x.name == l['left']['name']) {
+                const current_Domain = x;
```

Comment:
Instead of this, we should rename x to domain (so `for (const domain of responseDomains`)` above)